schemas/pci/pci-bus.yaml

```diff
@@ -22,139 +23,202 @@ description: |
 
 maintainers:
   - Rob Herring <robh@kernel.org>
+  - Pali Rohár <pali@kernel.org>
 
-properties:
-  $nodename:
-    pattern: "^pcie?@"
+$defs:
```

Comment:
I'm not sure using $defs is needed here and I rather avoid it if possible. Probably each def can be a separate file.

Reply:
I put all PCI stuff into `pci-bus.yaml` and all PCIe stuff in `pcie-bus.yaml`, just because all PCI/PCIe controllers use just that one top level schema. I defined more schemas in file just to split complicated hierarchy into logical parts and to easily define those recursive relations (like PCIe device is on the bus and PCIe bridge represents both device and another bus).

But if you think that splitting definitions into separate files is needed, I can do that. Just I thought that DTS files would use only top level schema, and so it would not be needed...

Reply:
From a binding perspective, there's not really a distinction between PCI and PCIe. There's an OF PCIe binding, but it's only a few extra properties which we don't even use.

In the end, we need a device specific schema we apply to a node (and generally only 1 level) which references 0 or more common schemas and has 'unevaluatedProperties: false'. You cannot have 2 schemas with disjoint sets of child node properties defined and have unevaluatedProperties work correctly as each child node schema is applied independently. This is why we now have spi-peripheral-props.yaml schema for example.